include/zephyr/net/tls_credentials.h

```diff
@@ -35,8 +35,11 @@ enum tls_credential_type {
 	TLS_CREDENTIAL_CA_CERTIFICATE,
 
 	/** A public server certificate. Use this to register your own server
-	 *  certificate. Should be registered together with a corresponding
-	 *  private key. Used with certificate-based ciphersuites.
+	 *  certificate, such as a certificate identifying the client device
```

Comment:
This still doesn't seem all that clear. The "SERVER" certificate here can be used as either a server certificate, or as a client certificate, determined by what kind of connection we have. Realistically, this should probably be renamed to something like TLS_CREDENTIAL_OWN_CERTIFICATE (along with TLS_CREDENTIAL_OWN_KEY), but changing the name will be a multi-faceted process.

Let's at least start by clearly stating what it is in the docs. This is the certificate, that along with the PRIVATE_KEY below defines the certificate to identify our end of the connection. If the socket is a server, then this will be the server certificate, if the socket is a client, then this will be the client connection.

Reply:
TLS_CREDENTIAL_LOCAL_CERTIFICATE? That seems clearer and avoids saying what the local role is … client or server.

Reply:
"own cert" comes from the implementation.

Reply:
I think `__DEPRECATED_MACRO` could be used:
https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/toolchain/gcc.h#L274